### PR TITLE
Prepare scrapy-splash 0.9.0

### LINKED ISSUE #307: Please Can we do a new release for all the changes merged in master since 8.0 in October'21
https://github.com/scrapy-plugins/scrapy-splash/compare/0.8.0...master

One that's nagging me is the following warning in my logs, for which the fix was applied in this PR https://github.com/scrapy-plugins/scrapy-splash/pull/269.

 if not major, can we please have a minor release 0.8.1 ?
```
[py.warnings] /usr/local/lib/python3.10/site-packages/scrapy_splash/[request.py:42](http://request.py:42/): ScrapyDeprecationWarning: Call to deprecated function to_native_str. Use to_unicode instead. Less

  url = to_native_str(url)
```

### PR DESCRIPTION
**Pull request by Gallaecio**
Resolves https://github.com/scrapy-plugins/scrapy-splash/issues/307

### CODE REVIEW

File: scrapy_splash/request.py [hunk 1]
```diff
@@ -90,13 +90,11 @@ def _original_url(self):
...
 
-    def __str__(self):
+    def __repr__(self):
         if not self._processed:
-            return super(SplashRequest, self).__str__()
+            return super().__repr__()
         return "<%s %s via %s>" % (self._original_method, self._original_url, self.url)
 
-    __repr__ = __str__
-
```
Comment (Gallaecio) [inline scrapy_splash/request.py:99]: The previous implementation caused a recursion error with modern Python versions.

File: setup.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 setup(
     name='scrapy-splash',
-    version='0.7.2',
+    version='0.8.0',
```
Comment (Gallaecio) [inline setup.py:6]: This is from merging the 0.8.0 branch into master. The actual version bump will be done as part of the release procedure, *after* we merge these changes.

File: tests/test_fingerprints.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 import pytest
 import scrapy
-from scrapy.dupefilters import request_fingerprint
+from scrapy.utils.request import request_fingerprint
```
Comment (Gallaecio) [inline tests/test_fingerprints.py:7]: We are still using a deprecated function, but supporting the new request fingerprinting may require more profound changes.

I wonder if maybe we can keep `request_fingerprint` around for longer than the minimum 1-year deprecation period on the Scrapy side.

File: scrapy_splash/response.py [hunk 4]
```diff
@@ -128,9 +128,6 @@ def data(self):
     def text(self):
         return self._ubody
 
-    def body_as_unicode(self):
-        return self._ubody
```
Comment (kmike) [inline scrapy_splash/response.py:132]: I think it makes sense to keep it, to avoid breaking user code. We can issue a deprecation warning.

**Comment by kmike** [comment]
Thanks @Gallaecio!

**Pull request merged**